platform/src/error_code.rs

```diff
@@ -23,6 +23,7 @@ pub enum ErrorCode {
     NoDevice = 11,
     Uninstalled = 12,
     NoAck = 13,
+    BadRVal = 1024,
```

Comment:
I don't think we need to assign `1024` to `BadRVal`, perhaps we can leave it without a specified discriminant?

I think we can remove the `TryFrom<u32> for ErrorCode` impl below, as it seems to be unused and I find it unlikely that I'll add a use case in the future.

Reply:
I strongly disagree - knowing specifically which integers map to which discriminants in the code is invaluable when debugging.

We do use `TryFrom<u32> for ErrorCode` occasionally in ti50, and it's useful there. I would prefer we keep it, considering I added it for us recently.

Reply:
Ah, I didn't remember that it was added for ti50.